CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift

```diff
@@ -54,31 +61,61 @@ public class AnalyticsManager {
 
         for property in properties {
             let propertyValue = property.value.analyticsPropertyValue()
-            if let propertyValue = propertyValue as? Bool {
-                eventProperties[property.key.rawValue] = analyticsString(from: propertyValue)
+            eventProperties[property.key.rawValue] = convertPropertyValueToString(propertyValue)
+        }
+
+        mixpanelInstance?.track(event: event.rawValue, properties: eventProperties)
+    }
+
+    public static func trackUserProperties(_ properties: [AnalyticsUserProperty: AnalyticsPropertyValue]) {
+        if mixpanelInstance != nil {
+            var propertiesToTrack: [String: String] = [:]
+
+            for (property, value) in properties {
+                propertiesToTrack[property.rawValue] = convertPropertyValueToString(value)
             }
 
-            if let propertyValue = propertyValue as? String {
-                eventProperties[property.key.rawValue] = propertyValue
+            mixpanelInstance?.people.set(properties: propertiesToTrack)
+        } else {
+            for (property, value) in properties {
+                userProperties[property] = value
             }
+        }
+    }
 
-            if let propertyValue = propertyValue as? Int {
-                eventProperties[property.key.rawValue] = "\(propertyValue)"
+    public static func registerSuperPropertiesOnce(_ properties: [AnalyticsSuperProperty: AnalyticsPropertyValue]) {
+        if mixpanelInstance != nil {
+            var superProperties: [String: String] = [:]
+
+            for (property, value) in properties {
+                superProperties[property.rawValue] = convertPropertyValueToString(value)
             }
 
-            if let propertyValue = propertyValue as? [String] {
-                eventProperties[property.key.rawValue] = arrayToString(from: propertyValue)
+            mixpanelInstance?.registerSuperPropertiesOnce(superProperties)
+        } else {
+            for (property, value) in properties {
+                superProperties[property] = value
             }
         }
+    }
```

Comment:
[suggestion]: I suggest extracting common code into a method like this `mapPropertiesToDictionary`
```
private static func mapPropertiesToDictionary(_ properties: [AnalyticsPropertyType: AnalyticsPropertyValue]) -> [String: String] {
    var dictionary: [String: String] = [:]

    for (property, value) in properties {
        dictionary[property.rawValue] = convertPropertyValueToString(value)
    }

    return dictionary
}
```
and reuse it in both `trackUserProperties` and `trackUserProperties`.

Reply:
@ValentinaIancu-Gini in that case I will be able to do it via generic, because we have 2 different enums for dict keys, not one `AnalyticsPropertyType`
Something like: `mapPropertiesToDictionary<T: RawRepresentable>(_ properties: [T: AnalyticsPropertyValue])`
Sounds good?

Reply:
if we want to include also
```
 for (property, value) in properties {
                superProperties[property] = value
            }
``` yes we need to use generics. 
I think is better like this than having duplicate-ish code.

Reply:
@ValentinaIancu-Gini I created a reusable func and moved both suggested code blocks there. Thanks for your suggestions!